### PR TITLE
Over/Out Events

### PR DESCRIPTION
**Pull request by AtheistP3ace**
Added Over/Out events

### CODE REVIEW

**Comment by AtheistP3ace** [comment]
I changed the names you suggest from enter/leave to over/out since I felt they made more sense. For instance the moment I start dragging an element it has really entered its container but it is over a droppable target and when I drop on a target to me out more represents there is no currently dragging item that is in a position to be dropped, which is why I also had the cleanup function emit out as well.

File: dragula.js [hunk 1]
```diff
@@ -301,6 +305,17 @@ function dragula (initialContainers, options) {
     var item = _copy || _item;
     var elementBehindCursor = getElementBehindPoint(_mirror, _clientX, _clientY);
     var dropTarget = findDropTarget(elementBehindCursor, _clientX, _clientY);
+    if (dropTarget !== null && !_emittedOver) {
+        _lastTargetOver = dropTarget;
+        _emittedOver = true;
+        // trigger over
```
Comment (bevacqua) [inline dragula.js]: these comments are unnecessary

File: dragula.js [hunk 2]
```diff
@@ -301,6 +305,17 @@ function dragula (initialContainers, options) {
...
+        api.emit('over', item, dropTarget, _source);
+    }
+    else if (dropTarget === null && _emittedOver) {
```
Comment (bevacqua) [inline dragula.js]: please stay with the existing coding style, move the `else` to the previous line

**Comment by bevacqua** [comment]
Sounds good to me. Please add the relevant documentation. Thanks

**Comment by AtheistP3ace** [comment]
Do you need me to make those changes in the PR code or are you?

**Comment by bevacqua** [comment]
Yeah please do

**Comment by AtheistP3ace** [comment]
OK, let me know if that covers everything. Not sure if documentation is just updating that read.markdown and I hit all the areas necessary.

**Comment by bevacqua** [comment]
Perfect, one last detail, squash your commits. You can do:

```shell
git reset HEAD~3
git commit -am "Added over/out events"
git push --force
```

**Comment by AtheistP3ace** [comment]
Sorry man, I did this all through github site. How do I execute those commands using the website? I never pulled my fork down to my local, made my changes there and used command line to push them back.

**Comment by AtheistP3ace** [comment]
Hey I think I got those commands to run. Let me know if everything is good.

**Comment by bevacqua** [comment]
I refactored what you did a little and added a demo. It's in release `2.1.0`

**Comment by AtheistP3ace** [comment]
Sweet. Thanks!

**Pull request merged**